power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -38,6 +39,101 @@ constexpr from_json_t from_json;
 
 namespace detail {
 
+using nlohmann::json;
+
+// visitor for json conversion
+struct JsonMapArrayData {
+    size_t size{};
+    msgpack::sbuffer buffer{};
+};
+
+struct JsonSAXVisitor {
+    msgpack::packer<msgpack::sbuffer> top_packer() {
+        if (data_buffers.empty()) {
+            throw SerializationError{"Json root should be a map!\n"};
+        }
+        return {data_buffers.top().buffer};
+    }
+
+    template <class T> bool pack_data(T&& val) {
+        top_packer().pack(val);
+        ++data_buffers.top().size;
+        return true;
+    }
+
+    bool null() {
+        top_packer().pack_nil();
+        ++data_buffers.top().size;
+        return true;
+    }
+    bool boolean(bool val) { return pack_data(val); }
+    bool number_integer(json::number_integer_t val) { return pack_data(val); }
+    bool number_unsigned(json::number_unsigned_t val) { return pack_data(val); }
+    bool number_float(json::number_float_t val, json::string_t const& /* s */) { return pack_data(val); }
+    bool string(json::string_t const& val) {
+        if (val == "inf" || val == "+inf") {
+            return pack_data(std::numeric_limits<double>::infinity());
+        }
+        if (val == "-inf") {
+            return pack_data(-std::numeric_limits<double>::infinity());
+        }
+        return pack_data(val);
+    }
+    bool key(json::string_t const& val) {
+        top_packer().pack(val);
+        return true;
+    }
+    static bool binary(json::binary_t const& /* val */) { return true; }
+
+    bool start_object(size_t /* elements */) {
+        data_buffers.emplace();
+        return true;
+    }
+    bool end_object() {
+        JsonMapArrayData object_data{std::move(data_buffers.top())};
+        data_buffers.pop();
```

Comment:
double-checking but i think if you `pop`, you get the value as well, right? (will check myself)

Reply:
nope, `pop` is `void`